workshop/content/intermediate/templates/resource-dependencies/_index.md

```diff
@@ -0,0 +1,273 @@
+{{% /notice %}}
+
+Kudos! You have now learned how CloudFormation automatically handles resource creation order when you define resource dependencies with `Ref` or `Fn::GetAtt`.
+
+
+### Challenge
+
+Design a template to create an [EC2 instance](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html), a [security group](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html) and an [S3 bucket](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-s3-bucket.html). Reference the security group logical ID in the EC2Instance resource for the `SecurityGroups` property. Your intent is to have CloudFormation initiate the S3 bucket resource creation only after the EC2 instance resource is created successfully. If you design the template correctly, as per the requirements, you should be able to observe stack events as follows:
+
+* Security Group resource creation is initiated.
+* Once the security group is marked as `CREATE_COMPLETE`, the EC2 instance resource creation starts.
+* After successful completion of the EC2 instance resource creation, CloudFormation should initiate the creation of S3 bucket.
+
+To get started, refer to the `resource-dependencies-challenge.yaml` template in the `code/workspace/resource-dependencies` directory: you will need to establish the resource dependencies as needed. Follow the above procedure to define the dependency, and verify stack events match the above series.
+
+{{%expand "Need a hint?" %}}
+
+* How can you [reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) the security group for your instance?
+* How can you [specify](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html) that the creation of a resource should follow another resource?
+{{% /expand %}}
+
+{{%expand "Need a solution?" %}}
+
+* Reference the security group resource LogicalID in EC2 instance resource properties as a list under `SecurityGroups` using the `Ref` intrinsic function. Doing this CloudFormation automatically waits for the security group to be created first, and then initiates the EC2 instance creation.
+* Modify the EC2 instance resource definition as shown next:
+
+```yaml
+  Ec2Instance:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: t2.micro
+      SecurityGroups:
+        - !Ref InstanceSecurityGroup
+```
+
+* Since there is no dependency between the EC2 instance and S3 bucket, use the `DependsOn` attribute, and provide the EC2 instance logical ID as a value for the `DependsOn` attribute.
```

Comment:
```suggestion
* Since there is no dependency between the EC2 instance and the S3 bucket, use the `DependsOn` attribute in the S3 bucket resource, and provide the EC2 instance's logical ID as a value for the `DependsOn` attribute.
```